release.nix

```diff
@@ -11,12 +11,89 @@
 let
   nixpkgs = import ./nixpkgs/17_09.nix;
 
-  config = {
-    allowUnfree = true;
+  config = { allowUnfree = true; };
 
-    packageOverrides = pkgs: {
-      haskellPackages = pkgs.haskellPackages.override {
-        overrides = self: super: rec {
+  overlays = [
+    (newPkgs: oldPkgs: {
+      haskellPackages = oldPkgs.haskellPackages.override {
+        overrides = newPkgs.lib.fold newPkgs.lib.composeExtensions (_: _: {}) [
+        
+        (newHaskellPkgs: oldHaskellPkgs: 
+
+         let
+
+           mk-proto3-suite = proto3-suite-boot:
+             newPkgs.haskell.lib.overrideCabal
+                (newHaskellPkgs.callPackage ./default.nix { })
+                (oldArgs:
+                  let
+                    python = newPkgs.python.withPackages (pkgs: [ pkgs.protobuf3_0 ]);
+
+                    test-files = ./test-files;
+
+                    cg-artifacts = newPkgs.runCommand "proto3-suite-test-cg-artifacts" {} ''
+                      mkdir -p $out/protos
+
+                      cp -r ${test-files}/. $out/protos/.
+
+                      cd $out
+
+                      build () {
+                        echo "[proto3-suite-test-cg-artifacts] Compiling proto-file/$1"
+                        ${proto3-suite-boot}/bin/compile-proto-file \
+                          --out $out \
+                          --includeDir "$2" \
+                          --proto "$1"
+                      }
+
+                      for proto in $(find ${test-files} -name 'test_*.proto'); do
+                        build ''${proto#${test-files}/} ${test-files}
+                      done
+
+                      echo "[proto3-suite-test-cg-artifacts] Protobuf CG complete"
+                    '';
+
+                  in rec {
+
+                     postPatch = (oldArgs.preCheck or "") + ''
+                       echo "Copying CG  artifacts from ${cg-artifacts} into ./gen/"
+                       mkdir -p gen
+                       ${newPkgs.rsync}/bin/rsync \
+                         --recursive \
+                         --checksum \
+                         ${cg-artifacts}/ gen
+                       chmod -R u+w gen
+                     '';
+
+                     shellHook = (oldArgs.shellHook or "") +
+                       (let
+                          ghc = newHaskellPkgs.ghcWithPackages (pkgs:
+                            oldArgs.testHaskellDepends ++ [ proto3-suite-boot ]
+                          );
+                        in ''
+                          ${postPatch}
+
+                          export PATH=${newHaskellPkgs.cabal-install}/bin:${ghc}/bin:${python}/bin''${PATH:+:}$PATH
+                        '');
+
+                     # We need to add dhall as a dependency because
```

Comment:
I think `cabal2nix` will correctly add the `dhall` dependency if you do `cabal2nix --flag dhall`

Reply:
Oh nice, I see it also generates a `configureFlags = ` definition...